Makefile

```diff
@@ -15,13 +15,33 @@
 # If you update this file, please follow
 # https://suva.sh/posts/well-documented-makefiles
 
+DBG ?= 0
+
+ifeq ($(DBG),1)
+GOGCFLAGS ?= -gcflags=all="-N -l"
+endif
+
+VERSION     ?= $(shell git describe --always --abbrev=7)
+REPO_PATH   ?= sigs.k8s.io/cluster-api-provider-azure
+LD_FLAGS    ?= -X $(REPO_PATH)/pkg/version.Raw=$(VERSION) -extldflags -static
+
 GO111MODULE = on
 export GO111MODULE
 GOFLAGS += -mod=vendor
 export GOFLAGS
 GOPROXY ?=
 export GOPROXY
 
+NO_DOCKER ?= 0
+ifeq ($(NO_DOCKER), 1)
+  DOCKER_CMD =
+  IMAGE_BUILD_CMD = imagebuilder
```

Comment:
I don't think this is being used, can probably drop it until it's required

Reply:
I talked about it with Alberto. Sometimes we want to run it in docker, but some commands might not work as expected so we need to revisit the functionality in future